lib/1.4/dml-builtins.dml

```diff
@@ -1812,9 +1863,17 @@ template bank is (object, shown_desc, _qname) {
         if (this._cached_bank_obj != NULL) {
             return this._cached_bank_obj;
         }
-        local char name[strlen(this._qname()) + 6];
-        strcpy(name, "bank.");
-        strcpy(name + 5, this._qname());
+        local const char *qn = this._qname();
+        local int len = strlen(qn);
+        local int anc_len;
+        for (anc_len = strlen(qn); anc_len > 0; --anc_len) {
+            if (qn[anc_len - 1] == '.')
+                break;
+        }
+        local char name[len + 6];
+        memcpy(name, qn, anc_len);
+        memcpy(name + anc_len, "bank.", 5);
+        strcpy(name + anc_len + 5, qn + anc_len);
```

Comment:
You've used this logic twice, now: break out the code.

Reply:
Right, probably better to do this in dmllib.h (also permits the use of stdio).

Reply:
Done now.